.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,180 @@
+name: Integration Test
+
+on:
+  pull_request:
+    paths-ignore:
+      - '**.md'
+  push:
+    branches:
+      - master
+
+env:
+  TIMEOUT: 15m
+  TYK_OSS_NS: tyk-oss
+  TYK_DATAPLANE_NS: tyk-dataplane
+  TYK_STACK_NS: tyk-stack
+
+jobs:
+  tyk-oss:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v4
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          check-latest: true
+      
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.6.0
+        
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.9.0
+        with:
+          node_image: "kindest/node:v1.29.1"
+      
+      - name: Deploy OSS Dependencies
+        run: |
+          kubectl create namespace ${{ env.TYK_OSS_NS }}
+          
+          # Do not change the name
+          helm install redis redis --wait --repo 'https://charts.bitnami.com/bitnami' -n ${{ env.TYK_OSS_NS }} --timeout ${{ env.TIMEOUT }} --version 19.0.2
+          
+          helm dependency update ./tyk-oss
+
+      - name: Run chart-testing (install)
+        run: ct install --namespace ${{ env.TYK_OSS_NS }} --charts ./tyk-oss --helm-extra-set-args "--wait=true" --helm-extra-set-args "--set=global.redis.addrs={redis-master.${{ env.TYK_OSS_NS }}.svc:6379},global.redis.passSecret.name=redis,global.redis.passSecret.keyName=redis-password"
+
+      - if: failure()
+        name: Check logs in case of failure
+        run: kubectl logs -n ${{ env.TYK_OSS_NS}} tyk-oss-test-tyk-oss
```

Comment:
i just realized that this line is not going to work because in the previous steps, we use `ct install` which gives arbitrary release name. therefore, the pod name does not exactly match `tyk-oss-test-tyk-oss`. Therefore, we cannot get the logs in case of failures.

Maybe, we could use `helm install` and `helm test` commands separately instead of `ct install`, as we did for tyk-stack. What do you think?